### ISSUE TITLE
RangeError: Invalid status code: undefined

### ISSUE THREAD

**Issue by Krillle** [SignalK/signalk-server-node#776]
Since server version 1.8.0 I get a RangeError when sending PUT commands via EmpirBus plugin. This causes a HTTP 500 error, while the command is executed nevertheless. Current running is 1.14.0 producing this log output:

```
Apr 16 16:17:58 SignalK signalk-server[11444]: RangeError: Invalid status code: undefined
Apr 16 16:17:58 SignalK signalk-server[11444]:     at ServerResponse.writeHead (_http_server.js:199:11)
Apr 16 16:17:58 SignalK signalk-server[11444]:     at ServerResponse.writeHead (/usr/lib/node_modules/signalk-server/node_modules/on-headers/index.js:44:26)
Apr 16 16:17:58 SignalK signalk-server[11444]:     at ServerResponse._implicitHeader (_http_server.js:190:8)
Apr 16 16:17:58 SignalK signalk-server[11444]:     at ServerResponse.end (/usr/lib/node_modules/signalk-server/node_modules/compression/index.js:103:14)
Apr 16 16:17:58 SignalK signalk-server[11444]:     at ServerResponse.send (/usr/lib/node_modules/signalk-server/node_modules/express/lib/response.js:221:10)
Apr 16 16:17:58 SignalK signalk-server[11444]:     at ServerResponse.json (/usr/lib/node_modules/signalk-server/node_modules/express/lib/response.js:267:15)
Apr 16 16:17:58 SignalK signalk-server[11444]:     at putPath.then.reply (/usr/lib/node_modules/signalk-server/lib/put.js:55:15)
Apr 16 16:17:58 SignalK signalk-server[11444]:     at <anonymous>
Apr 16 16:17:58 SignalK signalk-server[11444]:     at process._tickCallback (internal/process/next_tick.js:189:7)
Apr 16 16:17:58 SignalK signalk-server[11444]: PUT /signalk/v1/api/vessels/self/electrical/switches/empirBusNxt-instance14-switch3/state/ 500 6.798 ms - 30
```

**Comment by tkurki** [comment]
Is this still a problem and something to be fixed?

**Comment by Krillle** [comment]
I am still on version 1.8. I will update and report. It might take a while as boat is at land for winter. 

Do I have to consider major changes in configuration when updating from 1.8 to 1.18.1?

**Comment by tkurki** [comment]
No. Everything should be backwards compatible.

**Comment by Krillle** [comment]
I crosschecked with current installation 1.8. Problem seems to be gone:

```Dec 25 10:51:42 SignalK signalk-server[316]: GET /signalk/v1/api/vessels/self/tanks/freshWater/0/currentLevel/value 200 1471.051 ms - 1
Dec 25 10:51:42 SignalK signalk-server[316]: GET /signalk/v1/api/vessels/self/tanks/freshWater/0/currentLevel/value 200 1482.495 ms - 1
Dec 25 10:51:42 SignalK signalk-server[316]: GET /signalk/v1/api/vessels/self/electrical/switches/empirBusNxt-instance2-dimmer2/state/value 200 1510.438 ms - 5
Dec 25 10:51:42 SignalK signalk-server[316]: GET /signalk/v1/api/vessels/self/electrical/switches/empirBusNxt-instance18-switch5/state/value 200 1540.966 ms - 5
Dec 25 10:52:00 SignalK signalk-server[316]: PUT /signalk/v1/api/vessels/self/electrical/switches/empirBusNxt-instance16-switch4/state/ 200 12.670 ms - 168
Dec 25 10:52:12 SignalK signalk-server[316]: PUT /signalk/v1/api/vessels/self/electrical/switches/empirBusNxt-instance16-switch4/state/ 200 5.736 ms - 168```

**Comment by Krillle** [comment]
Issue solved in v1.16.0
